Library/OcStorageLib/OcStorageLib.c

```diff
@@ -253,30 +251,20 @@ OcStorageInitFromFs (
     return Status;
   }
 
+  SignatureSize = 0;
+
   if (StorageKey) {
     Signature = OcStorageReadFileUnicode (
       Context,
       OC_STORAGE_VAULT_SIGNATURE_PATH,
-      &DataSize
+      &SignatureSize
       );
 
     if (Signature == NULL) {
       DEBUG ((DEBUG_ERROR, "OCS: Missing vault signature\n"));
       OcStorageFree (Context);
       return EFI_SECURITY_VIOLATION;
     }
-
-    if (DataSize != CONFIG_RSA_KEY_SIZE) {
```

Comment:
So, we now cannot check this at early step but rather do it inside the cryptographic library? Is that good or bad? I am worried because of the valid signature algorithms once again.

Reply:
This was intentional, we *can* check it if we want of course, but probably just a minimum check. If the user wanted to use RSA-16384 or whatever, I don't see a reason to stop them. I took the old check as a limitation of the library design.

Reply:
Once again it it the same thing as an cipher whitelist, so it was not a limitation, but rather an agreement.

Reply:
I'll mark this as resolved as it's basically an implication of this discussion: https://github.com/acidanthera/OcSupportPkg/pull/18#discussion_r339431331